libs/auth/src/java_card.ts

```diff
@@ -440,7 +459,10 @@ export class JavaCard implements Card {
       );
     }
 
-    return cardDetails;
+    const numIncorrectPinAttempts = cardDoesNotHavePin
+      ? undefined
+      : (await this.getNumIncorrectPinAttempts()) || undefined;
```

Comment:
Rather than using a mix of `0` and `undefined` for the zero-case, I'm consistently using `undefined`.

I initially tried reserving `undefined` for cards without PINs and always defining a number, whether 0, for cards with PINs, but that added complexity to the types/code for little gain.